### PR TITLE
ci: add workflow to fetch contrib docs nightly

### PR DESCRIPTION
**Pull request by oddgrd**
I started implementing this and the workflow works, but I haven't gotten to the website parts yet.

The workflow checks out the website and hyper, lowercases all the files in hyper/docs and `cp`s them to `_contrib`. It then stages the changes, and if there are any staged changes it pushes a commit. Finally it pushes the commit (if there is one).

Workflow run: https://github.com/oddgrd/hyperium.github.io/actions/runs/3858409156/jobs/6576897139
Commit: https://github.com/oddgrd/hyperium.github.io/commit/3f258e358f3fc7ecf44fe1ae4d54aa4ef54011c2

Second run (no changes so no commit): https://github.com/oddgrd/hyperium.github.io/actions/runs/3858441335

For reference: https://github.com/actions/checkout#Push-a-commit-using-the-built-in-token

Closes https://github.com/hyperium/hyper/issues/3100

### CODE REVIEW

**Comment by seanmonstar** [comment]
Super cool! So, would this run on like a nightly basis? And, would it just automatically update the website? Or does it submit a PR we need to merge? Sorry if I missed a sentence explaining that...

**Comment by oddgrd** [comment]
Yes, it will run every night at 3am UTC (or if activated manually via cli/github UI), and it will automatically push a commit to master if there are any changes in `_contrib`.

I don't think it would be very hard to change it to send a PR instead, though. On one hand it makes me a little bit uncomfortable to have CI automatically push to master, on the other hand it is quite convenient. What would you prefer?

**Comment by seanmonstar** [comment]
I agree with both of your feelings: spooky, but would approving duplicate PRs the following day just be annoying? If it submitted PRs, and we didn't pay attention, or overeagerly dismissed the email notification, would that mean the website and main repo fall out of sync? Which do _you_ lean towards?

**Comment by oddgrd** [comment]
The current strategy *should* be safe since it only pushes changes to `_contrib`, but for a big project with many contributors it may cause problems down the line. I am leaning towards sending pull requests, and using an [action](https://github.com/marketplace/actions/create-or-update-pull-request) to overwrite it if it already exists (if we forgot to merge the previous one).

I think the optimal solution would involve a webhook in the hyper repo to trigger the website workflow on push to `hyper/docs`, so we'd get the updated PR immediately rather than at a somewhat arbitrary later time. Working across repos adds a bit of complexity though. 

I think we can start with the cron job PR version, if you're happy with that, and implementing the actual contrib section of the website. Then we can optimize it later with a web-hook. And when we've learned how to do this for the docs/contrib, we should be able to apply it easily to other sections that are duplicated between repos.

File: _contrib/commits.md
Comment (oddgrd) [inline _contrib/commits.md:4]: I updated the CI command to convert the titles to frontmatter, as well as replace '_' with '-'. The layout is tbd, not sure if I can reuse it yet or need a new one.

Comment (oddgrd) [reply _contrib/commits.md]: I did end up reusing the layout and I think it looks okay. The only thing missing now is fixing all the links, which could be a bit tricky since some of them use local urls and some of them use github urls. A PR to change all the links to local in the hyper repo may be needed, and then we can can just lowercase them in the CI workflow.

And also switching from the "commit to master" strategy to using PRs, but that should be easy.

File: _contrib/vision.md
Comment (oddgrd) [inline _contrib/vision.md:96]: Links like this seem to work in production, but not locally. :shrug: 

This is the error locally: `ERROR /contrib/vision/vision-arch.svg not found.`, and this is a working example that's live: https://github.com/hyperium/hyperium.github.io/blob/08c4c66d8fae807e9531d918f99cad097c98046b/_legacy/server/echo.md?plain=1#L8

File: _contrib/commits.md
Comment (seanmonstar) [reply _contrib/commits.md]: Local links makes sense to me.

**Comment by oddgrd** [comment]
Here is an example PR from the bot (with a dummy change): https://github.com/oddgrd/hyperium.github.io/pull/1

File: _contrib/governance.md
Comment (oddgrd) [inline _contrib/governance.md:113]: This is broken since in the hyper repo it links to the `CONTRIBUTING.md` doc in the root. An easy fix for this would be to make this a URL link instead.

File: _contrib/maintainers.md
Comment (oddgrd) [inline _contrib/maintainers.md:27]: The markdown in here doesn't apply, I could probably fix this with a CI command but it will add even more complexity.

File: _contrib/governance.md
Comment (oddgrd) [inline _contrib/governance.md:23]: I opted to also changes the names of the links to lowercase, but if you'd like to keep them capitalized (`[TENETS](./tenets.md)`) I'm happy to change it back (which will also reduce the CI complexity).

**Review by oddgrd — COMMENTED** [review]
This is essentially done @seanmonstar, except for a few minor details I left review comments for. 

The workflow did end up a little bit complicated, which I know is not ideal for maintainability. I'm happy to volunteer to maintain it though, at least for a good while, and hopefully find some clever ways to make it simpler in the long run.

There is also the question of implementing a trigger in the main repo so the update PR is sent immediately, but I think this is an optimization that can be applied later.

File: _contrib/maintainers.md
Comment (seanmonstar) [reply _contrib/maintainers.md:27]: Not a big deal.

File: _contrib/governance.md
Comment (seanmonstar) [reply _contrib/governance.md:23]: Seems fine being lowercased. Or uppercased, if that's easier. Either, really :)

File: _contrib/governance.md
Comment (seanmonstar) [reply _contrib/governance.md:113]: Huh, probably should figure out whether that page is worth importing... Eh, I don't want to block this PR on it.

**Review by seanmonstar — APPROVED** [review]
I think this looks fantastic. I'll wait till tomorrow I'm case there's anything you wanted to do before I click merge. Otherwise, 🎉!

File: _contrib/governance.md
Comment (oddgrd) [reply _contrib/governance.md]: That was very easy to do, so I just added it to the _contrib docs.

File: _contrib/governance.md
Comment (oddgrd) [reply _contrib/governance.md:23]: Cool, I'll leave it like this then for now.

**Pull request merged**
